pkg/router/template/router.go

```diff
@@ -657,13 +655,22 @@ func (r *templateRouter) DeleteServiceUnit(id ServiceUnitKey) {
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	_, ok := r.findMatchingServiceUnit(id)
+	service, ok := r.findMatchingServiceUnit(id)
 	if !ok {
 		return
 	}
 
+	r.DeleteServiceUnitInternal(id, service)
+}
+
+// DeleteServiceUnitUnitInternal deletes the service with the given
+// id. It differs from DeleteServiceUnitInternal() as it assumes the
+// caller has taken the lock.
+func (r *templateRouter) DeleteServiceUnitInternal(id ServiceUnitKey, service ServiceUnit) {
```

Comment:
Can `DeleteServiceUnitInternal` be private?

Reply:
> Can `DeleteServiceUnitInternal` be private?

No. But only because of the package naming, as we discovered yesterday.

Reply:
> > Can `DeleteServiceUnitInternal` be private?
> 
> No. But only because of the package naming, as we discovered yesterday.

I thought that was only true of `DeleteServiceUnit`.  It doesn't look like `DeleteServiceUnitInternal` is used outside of the file in which it is defined.

Reply:
Fixed in https://github.com/openshift/router/pull/243/commits/922203ce01bba78a2f07275c81cc29c35a85d568